bindings/nodejs/src/functions/uart.cc

```diff
@@ -140,36 +168,38 @@ NAN_METHOD(bind_sol_uart_write) {
         return;
 
     sol_uart *uart = uart_data->uart;
-    unsigned char *outputBuffer = (unsigned char *) 0;
-
     if (!node::Buffer::HasInstance(info[1])) {
         Nan::ThrowTypeError("Argument 1 must be a Buffer");
         return;
     }
 
+    struct sol_blob *blob;
     size_t length = node::Buffer::Length(info[1]);
-
-    outputBuffer = (unsigned char *) malloc(length * sizeof(unsigned char));
+    void *outputBuffer = malloc(length);
     if (!outputBuffer) {
         Nan::ThrowError("Failed to allocate memory for output buffer");
         return;
     }
 
     memcpy(outputBuffer, node::Buffer::Data(info[1]), length);
+    blob = sol_blob_new(&SOL_BLOB_TYPE_DEFAULT, NULL, outputBuffer, length);
```

Comment:
Instead of creating a new blob type, I will make the JS buffer as persistent and pass the buffer data to sol_blob_new(). Later when the feed callback is done, I will free the persistent JS buffer. With this simple approach I can avoid the copies of the JS buffer.